### PR TITLE
refactor: Refactor IndexService 

### PR DESCRIPTION
**Pull request by mykola-vrmchk**
### Motivation / Background

This Pull Request ...

### CODE REVIEW

**Comment by mykola-vrmchk** [comment]
* **#6391** <a href="https://app.graphite.dev/github/pr/dxos/dxos/6391?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @mykola-vrmchk and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/core/echo/indexing/src/query-service.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+          this._queries.delete(query);
+        },
+      };
+      this._queries.add(query);
+
+      queueMicrotask(async () => {
+        try {
+          const { changed } = await query.state.runQuery();
+          if (changed && !ctx.disposed) {
+            next({ queryId: request.queryId, results: query.state.getResults() });
```
Comment (dmaretskyi) [inline packages/core/echo/indexing/src/query-service.ts:96]: use `query.sendResults`

File: packages/core/echo/indexing/src/query-service.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+          }
+        } catch (error) {
+          log.catch(error);
```
Comment (dmaretskyi) [inline packages/core/echo/indexing/src/query-service.ts:99]: close stream with error instead

File: packages/core/echo/indexing/src/query-service.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+export class QueryServiceImpl extends Resource implements QueryService {
+  private readonly _queries = new Set<ActiveQuery>();
+
+  private readonly _updateQueries = new DeferredTask(this._ctx, async () => {
+    await Promise.all(
+      Array.from(this._queries).map(async (query) => {
+        const { changed } = await query.state.runQuery();
+        if (changed) {
+          query.sendResults(query.state.getResults());
```
Comment (dmaretskyi) [inline packages/core/echo/indexing/src/query-service.ts:43]: on error close stream with error

**Pull request merged**